lib/ansible/playbook/task.py

```diff
@@ -72,7 +72,7 @@ class Task(Base, Conditional, Taggable, CollectionSearch, Notifiable, Delegatabl
 
     async_val = NonInheritableFieldAttribute(isa='int', default=0, alias='async')
     changed_when = NonInheritableFieldAttribute(isa='list', default=list)
-    delay = NonInheritableFieldAttribute(isa='int', default=5)
+    delay = NonInheritableFieldAttribute(isa='float', default=5)
```

Comment:
Changes to existing field types should probably be in a separate PR from changes to field type handling, especially if we're changing to a field type that isn't getting any new behavior.

Reply:
agreed, i mostly did it to get ci_complete to find other issues, we already have tests and playbooks in the wild that assume it behaves as a float, so I'll push that sep